src/main/scala/mesosphere/marathon/core/volume/VolumesModule.scala

```diff
@@ -0,0 +1,94 @@
+package mesosphere.marathon.core.volume
+
+import com.wix.accord._
+import mesosphere.marathon.core.volume.providers._
+import mesosphere.marathon.state._
+
+/**
+  * VolumeProvider is an interface implemented by storage volume providers
+  */
+trait VolumeProvider[+T <: Volume] {
+  /** appValidation implements a provider's app validation rules */
+  val appValidation: Validator[AppDefinition]
+  /** groupValidation implements a provider's group validation rules */
+  val groupValidation: Validator[Group]
+
+  /** apply scrapes volumes from an application definition that are supported by this volume provider */
+  def collect(container: Container): Iterable[T]
+
+  val containerInjector: ContainerInjector[Volume]
```

Comment:
maybe I am wrong but could this be just a method that takes a `T` and a `CommandInfo.Builder` and applies the appropriate modifications to the `CommandInfo.Builder`?